packages/react-dom/src/events/plugins/__tests__/SimpleEventPlugin-test.js

```diff
@@ -361,12 +363,14 @@ describe('SimpleEventPlugin', function() {
       expect(button.textContent).toEqual('Count: 0');
 
       // Click the button many more times
-      click();
-      click();
-      click();
-      click();
-      click();
-      click();
+      await TestUtils.act(async () => {
+        click();
```

Comment:
This is the only one I'm not sure about. Actually I'm confused why it passes. The microtasks don't fire in between each click, so I would expect `this.state.count + 1` to always equal 1. What am I missing?

Reply:
Each click queues a microtask to synchronously flush the update, and none of those tasks are cancelled, so there are 7 updates when the microtask queue is flushed, right?

Reply:
Ah right. Could you update it to flush the microtasks in between? Since that's closer to what would actually happen in the browser.

Reply:
Also the fact that there are seven separate microtasks doesn't seem right. Should it see that there's already a task at the correct priority and then bail out?

Reply:
See: https://github.com/facebook/react/pull/20669/commits/0140bee698bae4ba06768e09451496285c5a82cd#r564921191

Reply:
^ link isn't working but it's a comment on this PR: https://github.com/facebook/react/pull/20669